src/common/test/componentMocking.tsx

```diff
@@ -119,17 +184,74 @@ export const renderProfileContextWrapper = async (
       () => {
         const element = getElement(selector);
         if (element && !value) {
-          if (selector.valueSelector) {
+          if (selector.valueSelector || element.nodeName === 'INPUT') {
             value = (element as HTMLInputElement).value;
+          } else {
+            value = element.textContent || undefined;
           }
-          value = element.textContent || undefined;
         }
       },
       { timeout: 150 }
     );
     return Promise.resolve(value);
   };
 
+  const triggerAction: TestTools['triggerAction'] = async selector => {
+    const button = getElement(selector);
+    fireEvent.click(button as Element);
+    return Promise.resolve();
+  };
+
+  const isDisabled: TestTools['isDisabled'] = element =>
+    !!element && element.getAttribute('disabled') !== null;
+
+  const submit: TestTools['submit'] = async ({
+    waitForOnSaveNotification,
+    waitForAfterSaveNotification,
+    skipDataCheck,
+  } = {}) => {
+    const previousDataChangeTime = getLastTime('dataUpdateTime');
+    const submitButton = renderResult.container.querySelectorAll(
+      'button[type="submit"]'
+    );
+    fireEvent.click(submitButton[0]);
+    if (waitForOnSaveNotification) {
+      await waitForElementAndValue(waitForOnSaveNotification);
+    }
+    await waitFor(() => {
+      if (!isDisabled(submitButton[0])) {
+        throw new Error('NOT DISABLED');
+      }
+    });
+    if (waitForAfterSaveNotification) {
+      await waitForElementAndValue(waitForAfterSaveNotification);
+    }
+    if (!skipDataCheck) {
+      await waitForDataChange(previousDataChangeTime);
+    }
+  };
+
+  const setInputValue: TestTools['setInputValue'] = async props => {
+    const { newValue, target, selector } = props;
+    const input = target || getElement(selector);
+    fireEvent.change(input as HTMLElement, { target: { value: newValue } });
+
+    return waitFor(
+      () => {
+        getElement({ valueSelector: newValue });
+      },
+      { timeout: 60 }
+    );
+  };
+
+  const waitForElementAndValue: TestTools['waitForElementAndValue'] = async props => {
+    const { selector, value } = props;
+    const elementValue = await getTextOrInputValue(selector);
```

Comment:
Minor improvement suggestion: The getTextOrInputValue function seems to be applied only inside this function, and also it is inside the return promise. I wonder if we could define it above this waitForElementAndValue function so it would be closer to usage?

Reply:
Referring to another re-ordering comment:
https://github.com/City-of-Helsinki/open-city-profile-ui/pull/154#discussion_r618158453